python/lib/core/dmod/core/common/protocols.py

```diff
@@ -56,6 +58,209 @@ class DescribableProtocol(typing.Protocol):
     description: str
 
 
+@typing.runtime_checkable
+class JobResultProtocol(typing.Protocol[T]):
```

Comment:
Good catch - it was from an earlier fight with typing since `Future` isn't generic (meaning it can't indicate what it's returning). That use case is no longer necessary, so I'm going to see about just straight up removing it.

Reply:
Upon further inspection, it's for typing and some absolutely awful implementation details from the vanilla objects that this serves as a protocol for.

Reply:
That's fair, any reason not to name it `Future` since the concepts are so similar?